core/node/node_framework/src/implementations/layers/consistency_checker.rs

```diff
@@ -0,0 +1,77 @@
+use zksync_core::consistency_checker::ConsistencyChecker;
+use zksync_types::Address;
+
+use crate::{
+    implementations::resources::{
+        eth_interface::EthInterfaceResource, healthcheck::AppHealthCheckResource,
+        l1_batch_commit_data_generator::L1BatchCommitDataGeneratorResource,
+        pools::MasterPoolResource,
+    },
+    service::{ServiceContext, StopReceiver},
+    task::Task,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+pub struct ConsistencyCheckerLayer {
+    diamond_proxy_addr: Address,
+}
+
+impl ConsistencyCheckerLayer {
+    pub fn new(diamond_proxy_addr: Address) -> ConsistencyCheckerLayer {
+        Self { diamond_proxy_addr }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for ConsistencyCheckerLayer {
+    fn layer_name(&self) -> &'static str {
+        "consistency_checker_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        // Get resources.
+        let l1_client = context.get_resource::<EthInterfaceResource>().await?.0;
+
+        let pool_resource = context.get_resource::<MasterPoolResource>().await?;
+        let singleton_pool = pool_resource.get_singleton().await?;
+
+        let l1_batch_commit_data_generator = context
+            .get_resource::<L1BatchCommitDataGeneratorResource>()
+            .await?
+            .0;
+
+        let consistency_checker = ConsistencyChecker::new(
+            l1_client,
+            10, // TODO (BFT-97): Make it a part of a proper EN config
```

Comment:
Probably should be a field on `ConsistencyCheckerLayer `, since it'll be coming from the EN config.
Also, `BFT-97` is already done, so I don't think it's appropriate to reference it here.
Could you create a task in QIT with caption "Support External Node config in the framework"?

Reply:
sure

Reply:
https://linear.app/matterlabs/issue/QIT-30/support-external-node-config-in-the-framework

@popzxc, please edit it if the description provided by me is incorrect